uxarray/helpers.py

```diff
@@ -556,3 +556,111 @@ def _normalize_in_place(node):
         raise RuntimeError("Input array should have a length of 3: [x, y, z]")
 
     return list(np.array(node) / np.linalg.norm(np.array(node), ord=2))
+
+
+@njit
+def _pt_within_gcr(pt_cart, gcr_end_pt_cart_1, gcr_end_pt_cart_2):
+    """Helper function to determine if a point lies within the interval of the great circle arc, the accuracy is
+    within 1.0e-12 The great circle arc will be represented as two ndarrays: gcr_end_pt_cart_1 and gcr_end_pt_cart_2
+    since numba lacks support for the nested array
+
+    Parameters
+    ----------
+    pt_cart : float ndArray
+        3D Cartesian Coordinates [x, y, z] for the input point
+    gcr_end_pt_cart_1 : float ndArray
+        3D Cartesian Coordinates list [x1, y1, z1], one of the end points of the great circle arc
+    gcr_end_pt_cart_2 : float ndArray
+    3D Cartesian Coordinates list [x2, y2, z2], another end points of the great circle arc
+
+    Returns
+    -------
+    bool
+        True if the point lies within the interval of the gcr, False otherwise.
+
+    """
+    cross = np.cross(gcr_end_pt_cart_1, gcr_end_pt_cart_2)
+    dot = cross[0] * pt_cart[0] + cross[1] * pt_cart[1] + cross[2] * pt_cart[2]
+    if np.absolute(dot - 0) > 1.0e-12:
```

Comment:
The subtraction is actually necessary here. Due to the arithmetic error, we might not be able to get the exact 0.0

Reply:
Can you explain this further? Wouldn't the comparison to `1e-12` handle the case where we are not able to get the exact 0.0?

Reply:
> Can you explain this further? Wouldn't the comparison to `1e-12` handle the case where we are not able to get the exact 0.0?

Oh, I see what you mean here, sorry. Yes it can be directly compared to the epsilon here